projects/usdai/index.js

```diff
@@ -0,0 +1,99 @@
+const abi = require("./abi.json");
+const { sumTokens2 } = require("../helper/unwrapLPs");
+
+const USDAI_CONTRACT = "0x0A1a1A107E45b7Ced86833863f482BC5f4ed82EF";
+const STAKED_USDAI_CONTRACT = "0x0B2b2B2076d95dda7817e785989fE353fe955ef9";
+const PROTOCOL_CONTRACTS = [USDAI_CONTRACT, STAKED_USDAI_CONTRACT];
+const WRAPPED_M_CONTRACT = "0x437cc33344a0B27A429f795ff6B469C72698B291";
+const MAX_UINT_128 = "0xffffffffffffffffffffffffffffffff";
+
+async function tvl(api) {
+  // Get all pools engaged by the protocol
+  const pools = await api.call({
+    target: STAKED_USDAI_CONTRACT,
+    abi: abi.pools,
+  });
+  const tokens = await api.multiCall({ abi: abi.currencyToken, calls: pools });
+  const ct = await api.multiCall({ abi: abi.collateralToken, calls: pools });
+
+  // Pool deposit tokens and collateral tokens
+  const ownerTokens = pools.map((pool, i) => [[tokens[i], ct[i]], pool]);
+  const balances = await sumTokens2({ api, ownerTokens, permitFailure: true });
+
+  // Balance of wrapped M tokens in the protocol contracts
+  const balanceOfWrappedM = await api.multiCall({
+    abi: 'erc20:balanceOf',
+    calls: PROTOCOL_CONTRACTS.map(contract => ({
+      target: WRAPPED_M_CONTRACT,
+      params: [contract]
+    }))
+  });
+
+  // Add wrapped M balances
+  balanceOfWrappedM.forEach((balance, i) => {
+    api.addToken(WRAPPED_M_CONTRACT, balance, { skipChain: true })
+  });
+
+  // Immediately claimable wrapped M tokens
+  const claimableWrappedM = await api.call({
+    target: STAKED_USDAI_CONTRACT,
+    abi: abi.claimableBaseYield // return value is scaled up by 10^12
+  }) / 10 ** 12; // scale down by 10^12 to match the decimals of the wrapped M token
+
+  // Add claimable USDai
+  api.addToken(USDAI_CONTRACT, claimableWrappedM, { skipChain: true })
+  
+  return balances;
+}
+
+async function borrowed(api) {
```

Comment:
for CDPs, we dont track borrowed as the borrowed token in minted by the protocol, but if this USDAI token is pegged to US dollar, we can track it under out stablecoin dashboard